### PR TITLE
Make it possible to disable ctrlf-mode locally in buffers

### PR DESCRIPTION
**Pull request by andersjohansson**
Do this by defining ctrlf-global-mode and let ctrlf-mode be an
ordinary buffer-local minor-mode

See this more as a suggestion, one way of solving problems like #52

----

#

### CODE REVIEW

File: ctrlf.el [hunk 1]
```diff
@@ -1147,6 +1146,12 @@ See `ctrlf-mode-bindings' to customize."
         (advice-remove #'minibuffer-message
                        #'ctrlf--minibuffer-message-condense)))))
 
+;;;###autoload
+(define-globalized-minor-mode ctrlf-global-mode ctrlf-mode ctrlf-mode-turn-on)
```
Comment (raxod502) [inline ctrlf.el]: This is not needed, calling a mode function with no arguments enables it unconditionally in modern Emacs versions.

```suggestion
(define-globalized-minor-mode ctrlf-global-mode ctrlf-mode)
```

**Review by raxod502 — CHANGES_REQUESTED** [review]
This is reasonable, but most people should be enabling `ctrlf-mode` globally. Can we make the global mode called `ctrlf-mode` and the local mode called `ctrlf-local-mode`?

**Comment by andersjohansson** [comment]
Yes, I also thought it would be more reasonable to keep `ctrlf-mode`as the global mode initially (this would also be backward-compatible for ctrlf). Then the most common convention in emacs is to call the give the global mode the `global` qualifier (very often as a prefix, breaking package namespacing) and not the other way around (though I did not find any specific recommendations for this in the documentation). But as the global mode is the primary expected way of using ctrlf, going with `ctrlf-mode`and `ctrlf-local-mode`seems fine to me.

File: ctrlf.el (hunk 1 above)
Comment (andersjohansson) [reply ctrlf.el]: Fixed. We still need to provide a function for turning on, so it is:
```
(define-globalized-minor-mode ctrlf-mode ctrlf-local-mode ctrlf-local-mode)
```

**Comment by andersjohansson** [comment]
I didn't update the readme now, since I was unsure if you wanted to do the blocklist thing as well. The only mode hook I know now where `(ctrlf-local-mode -1)` should be added is `pdf-isearch-minor-mode-hook`.

**Review by raxod502 — APPROVED** [review]
Looks good to me. I'd be happy to merge this with a small README update and CHANGELOG entry. You could also add the blacklist if you wanted, but it would probably make more sense to do that in a separate pull request.

**Comment by andersjohansson** [comment]
Hope this looks good!

**Comment by raxod502** [comment]
Commit https://github.com/raxod502/ctrlf/pull/53/commits/44155e3d52f99a2d99ded381aa639280ad96879e improves the formatting. Otherwise, looks good. Thanks!

**Review by raxod502 — APPROVED** [review]

**Pull request merged**